### PR TITLE
docs(sample): update WriteToDefaultStream sample to match best practices

### PR DESCRIPTION
**Pull request by gnanda**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Comment by stephaniewang526** [comment]
Please follow format `docs(sample): update/add...` for sample changes in the future. Thanks!

**Review by stephaniewang526 — APPROVED** [review]

**Comment by stephaniewang526** [comment]
@yirutang please take a look also

**Review by averikitsch — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java [hunk 1]
```diff
@@ -45,36 +51,119 @@ public static void runWriteToDefaultStream()
...
+        synchronized (this.parent.lock) {
+          if (this.parent.error == null) {
+            StorageException storageException = Exceptions.toStorageException(throwable);
+            this.parent.error =
+                (storageException != null) ? storageException : new RuntimeException(throwable);
+          }
         }
-        ApiFuture<AppendRowsResponse> future = writer.append(jsonArr);
-        AppendRowsResponse response = future.get();
+        System.out.format("Error: %s\n", throwable.toString());
```
Comment (yirutang) [inline samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java:159]: Could we add some simple immediate retry pattern here?

Comment (gnanda) [reply samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java]: Done

**Review by yirutang — APPROVED** [review]

**Pull request merged**